libgalois/src/PropertyManager.cpp

```diff
@@ -52,53 +52,48 @@ katana::PropertyManager::GetProperty(const katana::Uri& property_path) {
 
 void
 katana::PropertyManager::PropertyLoadedActive(
-    const std::shared_ptr<arrow::Table>& property) const {
+    const std::shared_ptr<arrow::Table>& property) {
   KATANA_LOG_DEBUG_ASSERT(property);
-  auto bytes = static_cast<count_t>(katana::ApproxTableMemUse(property));
-  MemorySupervisor::Get().BorrowActive(Name(), bytes);
+  auto sz = katana::ApproxTableMemUse(property);
+  stats.bytes_loaded += sz;
+  katana::GetTracer().GetActiveSpan().Log(
+      "property cache loaded active", {
+                                          {"name", property->field(0)->name()},
+                                          {"approx_size_gb", ToGB(sz)},
+                                      });
 }
 
 void
 katana::PropertyManager::PutProperty(
     const katana::Uri& property_path,
     const std::shared_ptr<arrow::Table>& property) {
   auto bytes = static_cast<count_t>(katana::ApproxTableMemUse(property));
-  auto granted = MemorySupervisor::Get().ActiveToStandby(Name(), bytes);
-  if (granted >= static_cast<count_t>(bytes)) {
-    cache_->Insert(property_path, property);
-    katana::GetTracer().GetActiveSpan().Log(
-        "property cache insert",
-        {
-            {"storage_name", property_path.BaseName()},
-            {"approx_size_gb", ToGB(katana::ApproxTableMemUse(property))},
-        });
-  } else {
-    MemorySupervisor::Get().ReturnActive(Name(), bytes);
-  }
+  cache_->Insert(property_path, property);
+  katana::GetTracer().GetActiveSpan().Log(
+      "property cache insert",
+      {
+          {"storage_name", property_path.BaseName()},
+          {"approx_size_gb", ToGB(katana::ApproxTableMemUse(property))},
+      });
+  MemorySupervisor::Get().ActiveToStandby(Name(), bytes);
 }
 
 katana::count_t
 katana::PropertyManager::FreeStandbyMemory(count_t goal) {
-  count_t total = 0;
   auto scope = katana::GetTracer().StartActiveSpan("free standby memory");
   scope.span().Log(
       "before", {
                     {"goal_gb", ToGB(goal)},
                     {"cache_gb", ToGB(cache_->size())},
                 });
 
-  if (goal >= static_cast<count_t>(cache_->size())) {
-    total = static_cast<count_t>(cache_->size());
-    MemorySupervisor::Get().ReturnStandby(Name(), std::min(goal, total));
-    cache_->clear();
-  } else {
-    total = static_cast<count_t>(cache_->Reclaim(goal));
-    MemorySupervisor::Get().ReturnStandby(Name(), std::min(goal, total));
-  }
+  auto reclaim = static_cast<count_t>(cache_->Reclaim(goal));
+  MemorySupervisor::Get().ReturnStandby(Name(), reclaim);
```

Comment:
Can the amount of standby memory change by the time we try to reclaim X bytes that are passed in? Or is that the point of `Reclaim` in the property cache?

Reply:
The code is single-threaded, so I don't anticipate the amount of standby memory changing.